Source/JavaScriptCore/wasm/js/WebAssemblyModuleRecord.cpp

```diff
@@ -534,6 +513,58 @@ void WebAssemblyModuleRecord::initializeExports(JSGlobalObject* globalObject)
         makeFunctionWrapper(functionIndexSpace);
 
+    // Tables
+    for (unsigned i = 0; i < moduleInformation.tableCount(); ++i) {
+        if (moduleInformation.tables[i].isImport()) {
+            // We should either have a Table import or we should have thrown an exception.
+            RELEASE_ASSERT(m_instance->table(i));
+        }
+
+        if (!m_instance->table(i)) {
+            RELEASE_ASSERT(!moduleInformation.tables[i].isImport());
+            // We create a Table when it's a Table definition.
+            RefPtr<Wasm::Table> wasmTable = Wasm::Table::tryCreate(moduleInformation.tables[i].initial(), moduleInformation.tables[i].maximum(), moduleInformation.tables[i].type(), moduleInformation.tables[i].wasmType());
+            if (!wasmTable)
+                return exception(createJSWebAssemblyLinkError(globalObject, vm, "couldn't create Table"_s));
+
+            Wasm::TableInformation::InitializationType initType = moduleInformation.tables[i].initType();
+            uint64_t initialBitsOrImportNumber = moduleInformation.tables[i].initialBitsOrImportNumber();
+            switch (initType) {
+            case Wasm::TableInformation::Default:
+                break;
+            case Wasm::TableInformation::FromGlobalImport:
+                ASSERT(initialBitsOrImportNumber < moduleInformation.firstInternalGlobal);
+                initialBitsOrImportNumber = m_instance->instance().loadI64Global(initialBitsOrImportNumber);
+                break;
+            case Wasm::TableInformation::FromRefFunc:
+                ASSERT(initialBitsOrImportNumber < moduleInformation.functionIndexSpaceSize());
+                ASSERT(makeFunctionWrapper(initialBitsOrImportNumber).isCallable());
+                initialBitsOrImportNumber = JSValue::encode(makeFunctionWrapper(initialBitsOrImportNumber));
+                break;
+            case Wasm::TableInformation::FromExtendedExpression:
+                ASSERT(initialBitsOrImportNumber < moduleInformation.constantExpressions.size());
+                evaluateConstantExpression(globalObject, moduleInformation.constantExpressions[initialBitsOrImportNumber], moduleInformation, moduleInformation.tables[i].wasmType(), initialBitsOrImportNumber);
+                RETURN_IF_EXCEPTION(scope, void());
+                break;
+            default:
+                ASSERT(initType == Wasm::TableInformation::FromRefNull);
+            }
+
+            JSWebAssemblyTable* table = JSWebAssemblyTable::tryCreate(globalObject, vm, globalObject->webAssemblyTableStructure(), wasmTable.releaseNonNull());
+            // We should always be able to allocate a JSWebAssemblyTable we've defined.
+            // If it's defined to be too large, we should have thrown a validation error.
+            scope.assertNoException();
+            ASSERT(table);
+            m_instance->setTable(vm, i, table);
+            RETURN_IF_EXCEPTION(scope, void());
```

Comment:
Should we crash instead?

Reply:
I looked into this and I think it's fine to not crash here, since the exception is checked by the callers of `initializeExports` and should error out soon after.